src/Common/LoggingFormatStringHelpers.h

```diff
@@ -113,12 +123,37 @@ template <typename T> constexpr std::string_view tryGetStaticFormatString(T && x
     }
 }
 
+void tryGetFormattedArgs(std::vector<std::string>&, bool) { }
+
+template <typename T> void tryGetFormattedArgs(std::vector<std::string>& out, bool ignoreFirst, T && x)
+{
+    if (!ignoreFirst)
+    {
+        if constexpr (std::is_base_of_v<fmt::detail::view, std::decay_t<T>>)
+        {
+            out.push_back(fmt::format("{}", std::remove_reference_t<T>(x)));
+        }
+        else
+        {
+            out.push_back(fmt::format("{}", std::forward<T>(x)));
```

Comment:
It will format arguments twice: the first time here and the second time inside fmtlib. It would be much more optimal to update the implementation if fmtlib to make it return `string_view`s to the formatted arguments inside the result string. Or, at least, make it return `std::vector<std::string>` from fmtlib to avoid formatting numbers and complex structures (with custom `fmt::formatter<>` specialization) twice